### PR TITLE
Fix oops backtrace logging, it should be structured

### PR DESCRIPTION
**Pull request by andrewvc**
https://github.com/elastic/logstash/pull/4723 Ported to 2.x/2.2

### CODE REVIEW

File: logstash-core/lib/logstash/agent.rb [hunk 1]
```diff
@@ -213,15 +207,14 @@ def execute
...
+    @logger.error I18n.t("logstash.agent.error", :error => e)
     if !config_test?
-      report I18n.t("logstash.agent.configtest-flag-information")
+      @logger.info I18n.t("logstash.agent.configtest-flag-information")
     end
     return 1
   rescue => e
     @logger.unsubscribe(stdout_logs) if show_startup_errors
-    report I18n.t("oops", :error => e)
-    report e.backtrace if @logger.debug? || $DEBUGLIST.include?("stacktrace")
```
Comment (suyograo) [inline logstash-core/lib/logstash/agent.rb]: @jsvd q: when was this `DEBUGLIST` introduced? This should be solved by #4548

**Comment by suyograo** [comment]
I am LGTM, but @jsvd should take a look at this, since he introduced `#report`

**Comment by andrewvc** [comment]
@suyograo agreed, will wait for @jsvd 's LGTM

File: logstash-core/lib/logstash/agent.rb (hunk 1 above)
Comment (jsvd) [inline logstash-core/lib/logstash/agent.rb]: DEBUGLIST has been around for quite some time now:

```
logstash (git)-[tags/v1.4.2] % grep DEBUGLIST lib/* -r
lib/logstash/agent.rb:    report e.backtrace if @logger.debug? || $DEBUGLIST.include?("stacktrace")
lib/logstash/monkeypatches-for-debugging.rb:if $DEBUGLIST.include?("require")
lib/logstash/runner.rb:$DEBUGLIST = (ENV["DEBUG"] || "").split(",")
lib/logstash/runner.rb:if ENV["PROFILE_BAD_LOG_CALLS"] || $DEBUGLIST.include?("log")
```

**Comment by jsvd** [comment]
LGTM

**Pull request closed without merging**